akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -510,9 +539,16 @@ class CircuitBreaker(scheduler: Scheduler, maxFailures: Int, callTimeout: Finite
      */
     override def _enter(): Unit = {
       set(System.nanoTime())
-      scheduler.scheduleOnce(resetTimeout) {
+      scheduler.scheduleOnce(currentResetTimeout) {
         attemptReset()
       }
+      val nextResetTimeout = currentResetTimeout * factor match {
+        case f: FiniteDuration ⇒ f
+        case _                 ⇒ currentResetTimeout
+      }
+
+      if (nextResetTimeout < 36500.days) // Duration overflow protection (< 100 years)
```

Comment:
I think we should have an upper bound parameter for the reset timeout, otherwise it can easily grow too much without going back to half-open in a reasonable time